pkg/operator/ceph/cluster/controller.go

```diff
@@ -396,49 +403,58 @@ func (c *ClusterController) reconcileCephCluster(clusterObj *cephv1.CephCluster,
 		logger.Infof("skipping orchestration for cluster object %q in namespace %q because its cleanup policy is set", clusterObj.Name, clusterObj.Namespace)
 		return nil
 	}
-
-	cluster, ok := c.clusterMap[clusterObj.Namespace]
+	var clustr *cluster
+	rawCluster, ok := c.clusterMap.Load(clusterObj.Namespace)
 	if !ok {
 		// It's a new cluster so let's populate the struct
-		cluster = newCluster(c.OpManagerCtx, clusterObj, c.context, ownerInfo, c.rookImage)
+		clustr = newCluster(c.OpManagerCtx, clusterObj, c.context, ownerInfo, c.rookImage)
+	} else {
+		clustr = rawCluster.(*cluster)
```

Comment:
It makes me sad that there isn't a generic version of sync.Map in the core.  Google found a couple of examples like https://pkg.go.dev/github.com/zolstein/sync-map but its probably not worth pulling in another dep for a single use.

Reply:
Agreed, wish we had a generic sync map as well. It is a small enough use case I figure I won't pull that in.